### ISSUE TITLE
clippy suggests std::foo with #[!no_std] being enabled

### ISSUE THREAD

**Issue by pacak** [rust-lang/rust-clippy#7858]
Lint name: clippy::manual_swap

I tried this code:

```rust
#![no_std]
 ....

        let t = self.0;
        self.0 = self.1;
        self.1 = t;

```

I expected to see this happen: clippy sits quiet

Instead, this happened: got a "warning: this looks like you are swapping `self.0` and `self.1` manually"
Proposed change won't compile.

### Meta

**Rust version (`rustc -Vv`):**
```
commit-hash: 09c42c45858d5f3aedfa670698275303a3d19afa
commit-date: 2021-10-18
host: x86_64-unknown-linux-gnu
release: 1.56.0
LLVM version: 13.0.0
```
